src/services/layout-service.ts

```diff
@@ -15,9 +15,9 @@ const createLayoutService = (
     x: isSnapshot ? snapshotDataPage.qWidth : Math.min(layout.qHyperCube.qSize.qcx, MAX_COLUMN_COUNT),
     y: isSnapshot ? snapshotDataPage.qHeight : Math.min(layout.qHyperCube.qSize.qcy, MAX_ROW_COUNT),
   };
-  const hasPseudoDimOnLeft = qEffectiveInterColumnSortOrder
-    .slice(0, qNoOfLeftDims)
-    .some((index) => index === PSEUDO_DIMENSION_INDEX);
+  const isLeftDimension = (dimensionInfoIndex: number) =>
+    qEffectiveInterColumnSortOrder.slice(0, qNoOfLeftDims).some((index) => index === dimensionInfoIndex);
```

Comment:
Consider moving `qEffectiveInterColumnSortOrder.slice(0, qNoOfLeftDims)` out of the function. There is no reason to slice the array on every function call.

Reply:
Done